lib/sqlalchemy/orm/events.py

```diff
@@ -2354,14 +2508,21 @@ def wrap(target, *arg, **kw):
         if propagate:
             manager = instrumentation.manager_of_class(target.class_)
 
-            for mgr in manager.subclass_managers(True):
+            for mgr in manager.subclass_managers(True):  # type: ignore [no-untyped-call] # noqa: E501
                 event_key.with_dispatch_target(mgr[target.key]).base_listen(
                     propagate=True
                 )
                 if active_history:
                     mgr[target.key].dispatch._active_history = True
 
-    def append(self, target, value, initiator, *, key=NO_KEY):
+    def append(
+        self,
+        target: _O,
+        value: _T,
+        initiator: Event,
+        *,
+        key: EventConstants = NO_KEY,
```

Comment:
**mike bayer** (zzzeek) wrote:

this is correct.

There's a change that's needed in sqlalchemy.orm.attributes to support this correctly which is at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4329 but that can remain independent of this.

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4326